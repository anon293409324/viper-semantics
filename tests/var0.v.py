def foo():
    x = 0
