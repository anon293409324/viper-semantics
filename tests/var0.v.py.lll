  $return(
    0,
    $lll(
      $seq(
        $seq($mstore( 28, $calldataload( 0)),
        $seq($mstore( 32, 1461501637330902918203684832716283019655932542976),
        $seq($mstore( 64, 170141183460469231731687303715884105727),
        $seq($mstore( 96, -170141183460469231731687303715884105728),
        $seq($mstore( 128, 1701411834604692317316873037158841057270000000000),
             $mstore( 160, -1701411834604692317316873037158841057280000000000)
        ))))),
        $if(
          $eq( $mload( 0), 3264763256 ),
          $seq($pass,
          $seq($assert( $iszero( $callvalue)),
          $seq($mstore( 320 , 0),
               $stop
          )))
        )
      ),
      0
    )
  )
