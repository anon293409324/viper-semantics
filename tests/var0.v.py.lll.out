<T> <k> . </k> <xevm> ListItem ( JUMP ( label ( 2 ) ) ) ListItem ( JUMPDEST ( label ( 1 ) ) ) ListItem ( PUSH ( 1 , 0  .Bytes ) ) ListItem ( CALLDATALOAD ) ListItem ( PUSH ( 1 , 28  .Bytes ) ) ListItem ( MSTORE ) ListItem ( PUSH ( 21 , 1  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  .Bytes ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ListItem ( PUSH ( 1 , 32  .Bytes ) ) ListItem ( MSTORE ) ListItem ( PUSH ( 16 , 127  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  .Bytes ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ListItem ( PUSH ( 1 , 64  .Bytes ) ) ListItem ( MSTORE ) ListItem ( PUSH ( 32 , 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 128  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  .Bytes ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ListItem ( PUSH ( 1 , 96  .Bytes ) ) ListItem ( MSTORE ) ListItem ( PUSH ( 21 , 1  ( 42  ( 5  ( 241  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 253  ( 171  ( 244  ( 28  ( 0  .Bytes ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ListItem ( PUSH ( 1 , 128  .Bytes ) ) ListItem ( MSTORE ) ListItem ( PUSH ( 32 , 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 255  ( 254  ( 213  ( 250  ( 14  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  ( 0  .Bytes ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ) ListItem ( PUSH ( 1 , 160  .Bytes ) ) ListItem ( MSTORE ) ListItem ( PUSH ( 4 , 194  ( 152  ( 85  ( 120  .Bytes ) ) ) ) ) ListItem ( PUSH ( 1 , 0  .Bytes ) ) ListItem ( MLOAD ) ListItem ( EQ ) ListItem ( ISZERO ) ListItem ( JUMPI ( label ( 3 ) ) ) ListItem ( CALLVALUE ) ListItem ( ISZERO ) ListItem ( ISZERO ) ListItem ( JUMPI ( pc ) ) ListItem ( PUSH ( 1 , 0  .Bytes ) ) ListItem ( PUSH ( 2 , 1  ( 64  .Bytes ) ) ) ListItem ( MSTORE ) ListItem ( STOP ) ListItem ( JUMPDEST ( label ( 3 ) ) ) ListItem ( JUMPDEST ( label ( 2 ) ) ) ListItem ( PUSH ( label ( 1 ) ) ) ListItem ( PUSH ( label ( 2 ) ) ) ListItem ( SUB ) ListItem ( PUSH ( label ( 1 ) ) ) ListItem ( PUSH ( 1 , 0  .Bytes ) ) ListItem ( CODECOPY ) ListItem ( PUSH ( label ( 1 ) ) ) ListItem ( PUSH ( label ( 2 ) ) ) ListItem ( SUB ) ListItem ( PUSH ( 1 , 0  .Bytes ) ) ListItem ( RETURN ) </xevm> <height> 0 </height> <with> .Map </with> </T>
